src/xcode/ENA/ENA/Source/Services/Risk/Provider/__tests__/RiskProviderTests.swift

```diff
@@ -9,6 +9,228 @@ import ExposureNotification
 // swiftlint:disable file_length
 // swiftlint:disable:next type_body_length
 final class RiskProviderTests: XCTestCase {
+	
+	func testGIVEN_RiskCalculation_WHEN_ENFRiskHighAndCheckinRiskLow_THEN_RiskConsumerReturnsRiskHigh() {
+		// GIVEN
+		let store = MockTestStore()
+		let client = ClientMock()
+		
+		let duration = DateComponents(day: 1)
+
+		store.enfRiskCalculationResult = nil
+
+		let config = RiskProvidingConfiguration(
+			exposureDetectionValidityDuration: duration,
+			exposureDetectionInterval: duration
+		)
+		let exposureDetectionDelegateStub = ExposureDetectionDelegateStub(result: .success([MutableENExposureWindow()]))
+		
+		let appConfig = CachedAppConfigurationMock(with: SAP_Internal_V2_ApplicationConfigurationIOS())
+
+		let eventStore = MockEventStore()
+		let traceWarningPackageDownload = TraceWarningPackageDownload(
+		   client: client,
+		   store: store,
+		   eventStore: eventStore
+	   )
+		let today = Calendar.utcCalendar.startOfDay(for: Date())
+		let riskLevelPerDateENF = [today: RiskLevel.high]
+		let riskLevelPerDateCheckin = [today: RiskLevel.low]
+		
+		let riskProvider = RiskProvider(
+			configuration: config,
+			store: store,
+			appConfigurationProvider: appConfig,
+			exposureManagerState: .init(authorized: true, enabled: true, status: .active),
+			enfRiskCalculation: ENFRiskCalculationFake(riskLevelPerDate: riskLevelPerDateENF),
+			checkinRiskCalculation: CheckinRiskCalculationFake(riskLevelPerDate: riskLevelPerDateCheckin),
+			keyPackageDownload: makeKeyPackageDownloadMock(with: store),
+			traceWarningPackageDownload: traceWarningPackageDownload,
+			exposureDetectionExecutor: exposureDetectionDelegateStub
+		)
+		
+		let consumer = RiskConsumer()
+		
+		let didCalculateRiskExpectation = expectation(description: "expect didCalculateRisk to be called")
+
+		let didFailCalculateRiskExpectation = expectation(description: "expect didFailCalculateRisk not to be called")
+		didFailCalculateRiskExpectation.isInverted = true
+
+		let didChangeActivityStateExpectation = expectation(description: "expect didChangeActivityState to be called 3 times")
+		didChangeActivityStateExpectation.expectedFulfillmentCount = 3
+
+		var risk: Risk?
+		consumer.didCalculateRisk = { calculatedRisk in
+			risk = calculatedRisk
+			didCalculateRiskExpectation.fulfill()
+		}
+		consumer.didFailCalculateRisk = { _ in
+			didFailCalculateRiskExpectation.fulfill()
+		}
+		consumer.didChangeActivityState = { _ in
+			didChangeActivityStateExpectation.fulfill()
+		}
+	
+		riskProvider.observeRisk(consumer)
+		
+		// WHEN
+		
```

Comment:
Small thing, could we remove the extra spacing after when and then?

Reply:
Sure about that? If I remove it, if looks like a comment just for the next line.  But its a comment for several following lines.